massband/rdf/utils.py

```
@@ -0,0 +1,390 @@
+    Args:
+        mol: RDKit molecule, which can contain explicit hydrogens.
+        smarts_or_smiles: SMARTS (e.g., "[F]") or SMILES with atom maps (e.g., "C1[C:1]OC(=[O:1])O1").
+        hydrogens: How to handle hydrogens in the final returned list.
+
+    Returns:
+        A single, flat list of unique integer atom indices matching the criteria.
+    """
+    patt = Chem.MolFromSmarts(smarts_or_smiles)
+
+    if not patt:
+        raise ValueError(f"Invalid SMARTS/SMILES: {smarts_or_smiles}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)